File: payment-service-app/nodejs-framework-payment-app/express-payment/src/domain/service/UserService.ts
import { User } from '../entity/User';
import { UserRepository } from '../repository/UserRepository';

const userRepo = new UserRepository();

export class UserService {
    async createUser (userData: Partial<User>) {
        return await userRepo.createUser(userData);
    }

    async getUserById(id: number) {
        return await userRepo.getUserById(id);
    }

    async getUsers() {
        return await userRepo.getUsers();
    }

    async updateUser (id: number, userData: Partial<User>) {
        return await userRepo.updateUser(id, userData);
    }

    async deleteUser (id: number) {
        return await userRepo.deleteUser(id);
    }
}

File: payment-service-app/microservice-payment-gateway/src/main/java/com/microservice/payment/service/UserService.java
package com.microservice.payment.service;

import com.microservice.payment.entity.User;
import com.microservice.payment.repository.UserRepository;
import jakarta.ws.rs.NotFoundException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService {
    @Autowired
    private UserRepository userRepository;

    public User createUser(User user) {
        return userRepository.save(user);
    }
    public User getUserById(Long id) {
        return userRepository.findById(id).orElseThrow(() -> new NotFoundException(String.valueOf(id)));
    }
}
